apm-agent-plugins/apm-jdbc-plugin/src/main/java/co/elastic/apm/agent/jdbc/helper/JdbcHelper.java

```diff
@@ -176,6 +176,27 @@ private ConnectionMetaData getConnectionMetaData(@Nullable Connection connection
         return connectionMetaData;
     }
 
+    @Nullable
+    private String safeGetCatalog(Connection connection) {
+        String catalog = null;
+        Class<?> type = connection.getClass();
+        Boolean supported = isSupported(JdbcFeature.CATALOG, type);
+        if (supported == Boolean.FALSE) {
+            return null;
+        }
+
+        try {
+            catalog = connection.getCatalog();
+            if (supported == null) {
+                markSupported(JdbcFeature.CATALOG, type);
+            }
```

Comment:
Here having a `null` catalog might just indicate that the current state of the connection does not have a selected catalog.
Thus it would be better to just mark the feature supported when it does not throw an `SQLException`.
```suggestion
            markSupported(JdbcFeature.CATALOG, type);
```